packages/xo-web/src/xo-app/vm/tab-disks.js

```diff
@@ -443,6 +451,15 @@ class AttachDisk extends Component {
     (check, srs, vdi) => check(srs[vdi.$SR])
   )
 
+  _checkVdi = createSelector(
+    createSelector(
+      () => this.props.allVbds,
+      vbds => groupBy(vbds, 'VDI')
+    ),
+    () => this.state.vdi,
+    (vbdsByVdi, vdi) => vdi != null && some(vbdsByVdi[vdi.id], 'attached')
```

Comment:
IMHO, we should also show warning for not `attached` VBDs because it will have a duplicated link between VM and VDI.

Reply:
Typically, the `VDI <-> VBD <-> VM` is unique.

Reply:
The VBD link is not unique and it's the issue. It have also a point in the issue to show duplicated links.

Reply:
The issue is to prevent user if the `VDI` is already attached. 
And there is a single VBD for each VM-VDI pair.
for `There have also a point in the issue for showing duplicated links.`:  it's for duplicate `VDI`.

Reply:
Are you sure there is a single VBD for each VM-VDI ?

Reply:
So, we have a different comprehension of the issue, i request @pdonias to clarify the need ^^

Reply:
For now, we'll show a warning only when the selected VDI is already attached to that VM in particular. Later, we may also add a warning if the VDI is already attached at all (if it's necessary).
But in any case, we never *prevent* the user from doing anything.